apps/emqx_cluster_link/src/emqx_cluster_link_router_syncer.erl

```diff
@@ -0,0 +1,603 @@
+                St#st{status = connecting};
+            {error, Reason} ->
+                ?SLOG(error, #{
+                    msg => "cluster_link_init_failed",
+                    reason => Reason,
+                    target_cluster => TargetCluster,
+                    actor => Actor
+                }),
+                _ = maybe_alarm(Reason, St),
+                St#st{error = Reason, status = disconnected}
+        end,
+    TRef = erlang:start_timer(?ACTOR_REINIT_TIMEOUT, self(), actor_reinit),
+    St1#st{actor_init_req_id = ReqId, actor_init_timer = TRef}.
+
+post_actor_init(
+    St = #st{client = ClientPid, target = TargetCluster, actor = Actor, incarnation = Incr},
+    NeedBootstrap
+) ->
+    ok = start_syncer(TargetCluster, Actor, Incr),
+    NSt = schedule_heartbeat(St#st{client = ClientPid}),
+    process_bootstrap(NSt, NeedBootstrap).
+
+handle_connect_error(Reason, St) ->
+    ?SLOG(error, #{
+        msg => "cluster_link_connection_failed",
+        reason => Reason,
+        target_cluster => St#st.target,
+        actor => St#st.actor
+    }),
+    TRef = erlang:start_timer(?RECONNECT_TIMEOUT, self(), reconnect),
+    _ = maybe_alarm(Reason, St),
+    St#st{reconnect_timer = TRef, error = Reason, status = disconnected}.
+
+handle_client_down(Reason, St = #st{target = TargetCluster, actor = Actor}) ->
+    ?SLOG(error, #{
+        msg => "cluster_link_connection_failed",
+        reason => Reason,
+        target_cluster => St#st.target,
+        actor => St#st.actor
+    }),
+    %% TODO: syncer may be already down due to one_for_all strategy
+    ok = suspend_syncer(TargetCluster, Actor),
+    _ = maybe_alarm(Reason, St),
+    NSt = cancel_heartbeat(St),
+    process_connect(NSt#st{client = undefined, error = Reason, status = connecting}).
+
+process_bootstrap(St = #st{bootstrapped = false}, _NeedBootstrap) ->
+    run_bootstrap(St);
+process_bootstrap(St = #st{bootstrapped = true}, NeedBootstrap) ->
+    case NeedBootstrap of
+        true ->
+            run_bootstrap(St);
+        false ->
+            process_bootstrapped(St)
+    end.
+
+process_heartbeat(St = #st{client = ClientPid, actor = Actor, incarnation = Incarnation}) ->
+    ok = publish_heartbeat(ClientPid, Actor, Incarnation),
+    schedule_heartbeat(St).
+
+schedule_heartbeat(St = #st{heartbeat_timer = undefined}) ->
+    Timeout = emqx_cluster_link_config:actor_heartbeat_interval(),
+    TRef = erlang:start_timer(Timeout, self(), heartbeat),
+    St#st{heartbeat_timer = TRef}.
+
+cancel_heartbeat(St = #st{heartbeat_timer = undefined}) ->
+    St;
+cancel_heartbeat(St = #st{heartbeat_timer = TRef}) ->
+    ok = emqx_utils:cancel_timer(TRef),
+    St#st{heartbeat_timer = undefined}.
+
+%% Bootstrapping.
+%% Responsible for transferring local routing table snapshot to the target
+%% cluster. Does so either during the initial startup or when MQTT connection
+%% is re-established with a clean session. Once bootstrapping is done, it
+%% opens the syncer.
```

Comment:
nit: opens -> activates

Reply:
Thanks, I will fix it in the next follow-up PR..